test/typed-function-references.wast

```diff
@@ -41,4 +41,31 @@
       )
     )
   )
+  (func $more-ref-types
```

Comment:
Can we port this test to lit while we're modifying it?

Reply:
I think it's better to do such porting separately. Porting it in this PR would make the diff less obvious.

I could create a new lit test for the new functions here I guess, though when we port this we'd want to merge them... but I don't feel strongly on that?

Reply:
+1 on porting separately. Actually it's not trivial to port all files in `test/` because it tests not only `wasm-opt` but also roundtripping through binary and dwarf info. I think it'd be better to be done separately.

Reply:
It's not trivial to port these tests because these tests test not a single thing but many things, including roundtripping. 

Also I think it's better for tests of a PR to have changes showing the what that PR does. If we port the whole test in the same PR, the whole test will be shown as the diff, and it is hard to figure out which part of the test have changed in the PR.

Reply:
I do agree we should find a way to make progress on porting these tests. Perhaps it could be part of the sheriff rotation? If the sheriff isn't busy with breakages perhaps they could port some tests.

Reply:
The various test configurations aren't a big issue IMO because you can add multiple RUN lines (e.g. one with --roundtrip and one without) and have the auto-update script take care of the rest. Personally I would be happier with gradual porting as we touch files than with having the sheriff work on porting because the gradual porting requires fewer context switches from other ongoing work.

Reply:
I agree it's good to put more efforts in gradual porting. But even when doing gradual porting I personally prefer to make it a separate PR, if that's not too difficult. I think that's easier for review too, because we can review only the test changes the functionality PR, and when we review the porting PR we are aware that the PR does not change any contents, so we can only check the `RUN` lines and a few other things.

Reply:
Right, ideally I would like a separate preliminary or follow-on PR porting the test whenever one of these tests is modified.